crates/wash-cli/src/ui/mod.rs

```diff
@@ -64,33 +64,57 @@ async fn ensure_washboard(version: &str, base_dir: PathBuf) -> Result<PathBuf> {
 }
 
 async fn download_washboard(version: &str, install_dir: &PathBuf) -> Result<()> {
-    let release_url = format!(
-        "https://github.com/wasmCloud/wasmCloud/releases/download/washboard-ui-{version}/washboard.tar.gz"
-    );
-
-    // Download tarball
-    let resp = tls::DEFAULT_REQWEST_CLIENT
-        .get(&release_url)
-        .send()
+    let urls = vec![
+        format!(
+            "https://github.com/wasmCloud/wasmCloud/releases/download/typescript%2Fapps%2Fwashboard-ui%2F{version}/washboard.tar.gz"
+        ),
+        format!(
+            "https://github.com/wasmCloud/wasmCloud/releases/download/washboard-ui-{version}/washboard.tar.gz"
+        ),
+    ];
+
+    let body = try_download_from_urls(&urls)
         .await
-        .context("failed to request washboard tarball")?;
+        .context("Failed to download washboard-ui assets")?;
 
-    if resp.status() != reqwest::StatusCode::OK {
-        bail!("failed to download washboard tarball: {}", resp.status());
-    }
-
-    let body = resp
-        .bytes()
-        .await
-        .context("failed to read bytes from washboard tarball")?;
+    eprintln!("Downloaded washboard-ui@{}", version);
 
     // Unpack and copy to install dir
     let cursor = Cursor::new(body);
     let mut tarball = Archive::new(Box::new(GzipDecoder::new(cursor)));
     tarball
         .unpack(install_dir)
         .await
-        .context("failed to unpack washboard tarball")?;
+        .context("Failed to unpack washboard-ui assets")?;
 
     Ok(())
 }
+
+async fn try_download_from_urls(urls: &[String]) -> Result<bytes::Bytes> {
+    let mut last_error = None;
+
+    for url in urls {
+        match try_download(url).await {
+            Ok(body) => return Ok(body),
+            Err(e) => last_error = Some(e),
+        }
+    }
+
+    Err(last_error.unwrap_or_else(|| anyhow::anyhow!("Failed to find suitable download URL")))
+}
+
+async fn try_download(url: &String) -> Result<bytes::Bytes> {
+    let resp = tls::DEFAULT_REQWEST_CLIENT
```

Comment:
Wow,
 
I do have a really good question here 😄 (someone asked my like an hour ago) could you please use this client instead of `tls::DEFAULT_REQWEST_CLIENT` [this one with magical proxy things in it](https://github.com/wasmCloud/wasmCloud/blob/fc0aea1aa752dcd24d80e011e811b0d0e5a404ac/crates/wash-lib/src/start/github.rs#L95)? Should be a drop-in replacement.


Btw after my pr we can even go ahead and try to fetch the latest "any" version of the washboard too **later**. (this [PR](https://github.com/wasmCloud/wasmCloud/pull/2527) with a bit of modifications)

Reply:
updated against main with your changes in, @markkovari!

Reply:
Thank you